### PR TITLE
Fix kinematics checkJoints not returning false when outside limits

### PR DESCRIPTION
**Pull request by Levi-Armstrong**

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
@haudren Do you mind reviewing?

**Comment by Levi-Armstrong** [comment]
This addresses issue #551

File: tesseract_kinematics/test/kinematics_test_utils.h [hunk 1]
```diff
@@ -639,7 +639,7 @@ inline void runActiveLinkNamesIIWATest(const tesseract_kinematics::ForwardKinema
 inline void runActiveLinkNamesIIWATest(const tesseract_kinematics::InverseKinematics& kin)
 {
   EXPECT_FALSE(kin.checkJoints(Eigen::VectorXd::Zero(8)));
-  EXPECT_TRUE(kin.checkJoints(Eigen::VectorXd::Constant(7, std::numeric_limits<double>::max())));
+  EXPECT_FALSE(kin.checkJoints(Eigen::VectorXd::Constant(7, std::numeric_limits<double>::max())));
```
Comment (mpowelson) [inline tesseract_kinematics/test/kinematics_test_utils.h:642]: Perhaps make one negative max?

**Review by mpowelson — APPROVED** [review]

File: tesseract_kinematics/test/kinematics_test_utils.h (hunk 1 above)
Comment (Levi-Armstrong) [reply tesseract_kinematics/test/kinematics_test_utils.h:643]: Yep. It has been added.

**Review by haudren — COMMENTED** [review]
:+1: I just have one comment, this checkJoint function does not seem to leverage `isWithinLimits`, is this expected?

**Comment by Levi-Armstrong** [comment]
The only reason I can see is to provide the joint name for the joint outside the limits.

**Pull request merged**